core/lib/zksync_core/src/state_keeper/seal_criteria/mod.rs

```diff
@@ -111,6 +112,7 @@ pub(super) trait SealCriterion: fmt::Debug + Send + 'static {
         tx_count: usize,
         block_data: &SealData,
         tx_data: &SealData,
+        protocol_version: ProtocolVersionId,
```

Comment:
Having to pass `protovol_version ` everywhere is a little awkward. I feel like it's worth to discuss a little if there are other approaches. If we are not mindful about this, we might en d up having a `protocol_version` expanding to every method in our codebase. 

@popzxc @slowli - any quick thoughts on this?

Reply:
could we instead have it as part of the `StateKeeperConfig`, initially i tried to keep it out of there because i thought it was protocol version agnostic but seeing `bootloader_hash` and `default_aa_hash`, i'm thinking it has some reliance on protocol version. wdyt?

Reply:
`StateKeeperConfig` is static, meaning it's loaded from env at the start and isn't changed, whilst protocol version can be changed. `bootloader_hash` and `default_aa_hash` are artifacts, hopefully, they are not used anywhere and should be removed.

As for Roman's concern: I generally agree, maybe making seal criteria depend on protocol version is not the best solution. But we can remove this dependency after the upgrade, so it doesn't look that bad

Reply:
Agree with Misha (though I'd mark it with a TODO to actually remove it one day).

Reply:
In general, we could have the set of criteria used for a certain input be dependent on the protocol version (e.g., by extending the `SealCriterion` interface). In this way, dependency on the protocol version would be contained in `ConditionalSealer`. I'm not sure whether this would make things better or worse, given that it seems like this dependency on the protocol version isn't supposed to be permanent.